Master CI failing: mypy type error in DatePicker initial_date assignment

## Bug

`slackblocks/elements.py:258` triggers a mypy error on every master push since PR #138:

```
slackblocks/elements.py:258: error: Incompatible types in assignment
(expression has type "None", variable has type "str")  [assignment]
```

In the `DatePicker.__init__` fix from #135 the `else` branch sets `self.initial_date = None`, but mypy infers `str` from the `if` branch because no type annotation was attached. The merge bypassed this check with `--admin` and the failure persisted.

## Fix

Annotate `self.initial_date: Optional[str]` so both `str` and `None` assignments are valid.

--- a/slackblocks/elements.py
+++ b/slackblocks/elements.py
@@ -250,12 +250,11 @@ class DatePicker(Element):
     ) -> None:
         super().__init__(type_=ElementType.DATE_PICKER)
         self.action_id = validate_action_id(action_id)
+        self.initial_date: Optional[str] = None
         if initial_date:
             self.initial_date = datetime.strptime(initial_date, "%Y-%m-%d").strftime(
                 "%Y-%m-%d"
             )
-        else:
-            self.initial_date = None
         self.confirm = confirm
         self.focus_on_load = focus_on_load
         self.placeholder = Text.to_text(

